sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -78,6 +78,7 @@ class SQLContext private[sql](val sparkSession: SparkSession)
 
   private[sql] def sessionState: SessionState = sparkSession.sessionState
   private[sql] def sharedState: SharedState = sparkSession.sharedState
+  @deprecated("Use SparkSession.sessionState.conf instead", "4.0.0")
   private[sql] def conf: SQLConf = sessionState.conf
```

Comment:
This is not a public API, can it be deleted directly?

Reply:
It depends. You are probably going to break someone's code if you remove it.

Reply:
Yes, I'm also concerned that removing it would have negative implications, even though it's just an internal API of Spark.